index.mjs

```diff
@@ -146,23 +143,34 @@ ParallelHook.prototype[$map] = function(f){
   return ParallelHook(map(f)(this.hook));
 }
 
+const crash = e => Future(() => { throw e });
+
 ParallelHook.prototype[$ap] = function(mf){
   return ParallelHook(Hook(c => {
-    let f, x, b = false;
-    const mirror = cache(never);
```

Comment:
The trick with `cache(never)` wasn't actually the issue; but I replaced this approach with a more succinct one.